src/rust/ensogl/src/display/scene.rs

```diff
@@ -120,23 +120,63 @@ pub enum Target {
 }
 
 impl Target {
+
+    /// Decode the symbol_id and instance_id that was encoded in the `fragment_runner`.
+    ///
+    /// See the `encode` method for more information on the encoding.
+    fn decode(pack1:u32, pack2:u32, pack3:u32) -> (u32,u32) {
+        let value1 = (pack1 << 4) + (pack2 >> 4);
+        let value2 = pack3 + ((pack2 & 0x000F) << 8);
+        (value1, value2)
+    }
```

Comment:
a very minor thing - I would rename `pack` to `chunk` to better illustrate what it is about.

Reply:
Plus, encode provides proper docs here. In such a case, move `decode` under `encode` so when reading the code it is easier to first read proper docs :)